### DISCUSSION TITLE
is chakra-ui 1.4 compatible with < typescript4?

### DISCUSSION THREAD

**Discussion by rnitta** [chakra-ui/chakra-ui#3686] [Q&A]
with typescript3.9

```
$ tsc --noEmit
node_modules/@chakra-ui/menu/node_modules/@chakra-ui/popper/dist/types/use-popper.d.ts:4:16 - error TS1005: ',' expected.

4     offset?: [x: number, y: number];
                 ~

node_modules/@chakra-ui/menu/node_modules/@chakra-ui/popper/dist/types/use-popper.d.ts:4:27 - error TS1005: ',' expected.

4     offset?: [x: number, y: number];
                            ~


Found 2 errors.
```

this seems [Labeled Tuple Elements (ts4.0)](https://devblogs.microsoft.com/typescript/announcing-typescript-4-0/#labeled-tuple-elements)
I'm using ts3.9, so failed to parse this syntax.
 ts>=4 is required to use chakra-ui?

**Comment by chasinhues** [comment]
Looks like Chakra has a dependency on ^4.1.3. I'd suggest upgrading your TS version.

https://github.com/chakra-ui/chakra-ui/blob/6bdcdcdce192313504f9f73526a253a41d76a648/package.json#L167

I did see that [our docs state a minimum TS version of 3.8.0](https://chakra-ui.com/docs/getting-started#notes-on-typescript-) so we should need to update that.

**Reply by rnitta** [reply]
thanks.

**Reply by minardimedia** [reply]
Using `"typescript": "4.3.3"` the same error is showing

**Reply by whoisryosuke** [reply]
Having the same issue on 4.1.3. Using `skipLibCheck` and even ignoring `node_modules` in the `tsconfig`. No dice, site won't build because of the issue.

It stopped happening for a while then started happening again recently (I think during a node_modules wipe + fresh yarn install).

**Reply by whoisryosuke** [reply]
Was able to fix by updating to the latest version of Chakra UI (@chakra-ui/icons@1.0.15, @chakra-ui/react@1.6.6, @chakra-ui/system@1.7.2, @chakra-ui/theme@1.10.0) and Typescript (3.9.7).
